library/goptions.ml

```diff
@@ -289,18 +295,22 @@ let declare_int_option =
   declare_option
     (fun v -> IntValue v)
     (function IntValue v -> v | _ -> anomaly (Pp.str "async_option"))
+    (fun _ _ -> anomaly (Pp.str "async_option"))
 let declare_bool_option =
   declare_option
     (fun v -> BoolValue v)
     (function BoolValue v -> v | _ -> anomaly (Pp.str "async_option"))
+    (fun _ _ -> anomaly (Pp.str "async_option"))
 let declare_string_option =
   declare_option
     (fun v -> StringValue v)
     (function StringValue v -> v | _ -> anomaly (Pp.str "async_option"))
+    (fun x y -> x^" "^y)
```

Comment:
Sure, I don't mind. But I am not sure to understand what your worry is. The only impact the hardcoded character has (for warnings at least) is how the string will get printed by the `Print` command if the user ever types it. Other than that, the user will never be made aware of which character was internally used. So we should just choose a character that conveys well the idea of separator and that leads to a readable string. So it could be space, it could be comma, it could be comma-space, it could be semi-colon, and so on.